### ISSUE TITLE
Type for optional array element not hyperlinked

### ISSUE THREAD

**Issue by techfg** [typedoc2md/typedoc-plugin-markdown#719]
### What package is the bug related to?

typedoc-plugin-markdown

### Describe the issue

For a function that accepts an array of optional elements, the type of the element is not hyperlinked in the generated markdown doc.

For example, given:

```ts
export type Stuff = { a: string; b: string };
export const handleArrayOfOptionalStuff = (a: [Stuff?]) => {};
```

The generated markdown will be:

```text
• **a**: [`Stuff?`]
```

Repro: https://stackblitz.com/edit/vitejs-vite-nypddu

**Steps to Reproduce**
1. Open [repro](https://stackblitz.com/edit/vitejs-vite-nypddu)
2. `npm run docs:md`
3. View files in docs-md/functions

**Actual Result**
Generated markdown "links" the type `Stuff` for `handleStuff`, `handleOptionalStuff`, `handleArrayOfStuff` but not for `handleArrayOfOptionalStuff`

**Additional Information**
1. The issue does not occur when generating html files (`npm run docs:html`) 
2. The issue does not occur for optional parameters, only arrays with optional elements

### TypeDoc configuration

typedoc: 0.26.11
typedoc-plugin-markdown: 4.2.10
typescript" 5.6.3

### Expected behavior

The generated markdown should contain a hyperlink to `Stuff`:

```text
• **a**: [[`Stuff`](../type-aliases/Stuff.md)]
```

**Comment by tgreyuk** [comment]
Thanks for detailed description. 

Fix in typedoc-plugin-markdown@4.3.0

<img width="575" alt="Screenshot 2024-11-27 at 18 29 00" src="https://github.com/user-attachments/assets/2814aa35-6e3e-4e57-94bb-dc976692eab9">

https://github.com/typedoc2md/typedoc-plugin-markdown-scratchpad/blob/main/issues/719/docs/md/functions/handleArrayOfStuff.md

Please note parameters layout has been tweaked in listing view to make a bit clearer. For a concise view parametersFormat-table can be used.

**Comment by techfg** [comment]
Hi @tgreyuk -

Thanks for the quick response on this, greatly appreciated!

In the repro provided, all looks good!  Unfortunately, when the type includes union and the entire union itself is optional, the docs do not emit correctly.

Repro: https://stackblitz.com/edit/vitejs-vite-vyxrtv?file=src%2Findex.ts

TypeDoc configuration
typedoc: 0.27.0
typedoc-plugin-markdown: 4.3.0
typescript" 5.7.2

```ts
export const handleArrayOfOptionalStuff = (
  a: [(null | undefined | Stuff)?]
) => {};
```

**Actual Result**
![image](https://github.com/user-attachments/assets/37242188-a0e5-48ae-a688-75715b4b63a1)

```text
### a

[`null` \| [`Stuff`](../type-aliases/Stuff.md)?]
```

**Expected Result**
![image](https://github.com/user-attachments/assets/9f7db4d6-545e-49e9-aabc-0ca2cd231615)

```text
### a

[(`null` \| [`Stuff`](../type-aliases/Stuff.md))?]
```

**Actual HTML Result**
![image](https://github.com/user-attachments/assets/97682987-6e8b-4ceb-b806-04432f85af21)

**Comment by tgreyuk** [comment]
Fix in typedoc-plugin-markdown@4.3.1

<img width="628" alt="Screenshot 2024-12-01 at 23 14 33" src="https://github.com/user-attachments/assets/bb87d65c-4a28-4a0e-bff5-535b3975dfb3">

**Comment by techfg** [comment]
Looks great in 4.3.1, thank you @tgreyuk!
